NEXTCLOUD.cmake

```diff
@@ -15,6 +15,62 @@ set( APPLICATION_VIRTUALFILE_SUFFIX "nextcloud" CACHE STRING "Virtual file suffi
+endif()
+if (DAILY_BUILD)
+    string(APPEND APPLICATION_NAME " (Daily Build)")
+    string(APPEND APPLICATION_SHORTNAME "-Daily")
+    string(APPEND APPLICATION_EXECUTABLE "-daily")
+    string(APPEND APPLICATION_ICON_NAME "Daily" )
+    string(APPEND APPLICATION_REV_DOMAIN "-daily")
+    string(APPEND LINUX_PACKAGE_SHORTNAME "-daily")
+    string(APPEND LINUX_APPLICATION_ID "-daily")
+    message(STATUS "DAILY_BUILD is set to ${DAILY_BUILD}, and name variables will be appended to prevent conflicts.")
+    message(NOTICE "   Set DAILY_BUILD to False or OFF to disable the DAILY_BUILD suffix.")
+    message(NOTICE "")
+    message(NOTICE "   Your choice will be cached. For further explanation, see:")
+    message(NOTICE "   https://cmake.org/cmake/help/latest/command/option.html")
+    message(NOTICE "")
+    message(NOTICE " * APPLICATION_NAME           is now:   \"${APPLICATION_NAME}\"")
+    message(NOTICE " * APPLICATION_SHORTNAME      is now:   \"${APPLICATION_SHORTNAME}\"")
+    message(NOTICE " * APPLICATION_EXECUTABLE     is now:   \"${APPLICATION_EXECUTABLE}\"")
+    message(NOTICE " * APPLICATION_ICON_NAME      is now:   \"${APPLICATION_ICON_NAME}\"")
+    message(NOTICE " * APPLICATION_REV_DOMAIN     is now:   \"${APPLICATION_REV_DOMAIN}\"")
+    message(NOTICE " * LINUX_PACKAGE_SHORTNAME    is now:   \"${LINUX_PACKAGE_SHORTNAME}\"")
+    message(NOTICE " * LINUX_APPLICATION_ID       is now:   \"${LINUX_APPLICATION_ID}\"")
+    message(NOTICE "")
+endif()
+if (TEST_BUILD AND NOT DAILY_BUILD)
+    string(APPEND APPLICATION_NAME " (Test Build)")
+    string(APPEND APPLICATION_SHORTNAME "-Test")
+    string(APPEND APPLICATION_EXECUTABLE "-test")
+    string(APPEND APPLICATION_ICON_NAME "Test" )
+    string(APPEND APPLICATION_REV_DOMAIN "-test")
+    string(APPEND LINUX_PACKAGE_SHORTNAME "-test")
+    string(APPEND LINUX_APPLICATION_ID "-test")
+    message(STATUS "TEST_BUILD is set to ${TEST_BUILD}, and name variables will be appended to prevent conflicts.")
+    message(NOTICE "   Set TEST_BUILD to False or OFF to disable the TEST_BUILD suffix.")
+    message(NOTICE "")
+    message(NOTICE "   Your choice will be cached. For further explanation, see:")
+    message(NOTICE "   https://cmake.org/cmake/help/latest/command/option.html")
+    message(NOTICE "")
+    message(NOTICE " * APPLICATION_NAME           is now:   \"${APPLICATION_NAME}\"")
+    message(NOTICE " * APPLICATION_SHORTNAME      is now:   \"${APPLICATION_SHORTNAME}\"")
+    message(NOTICE " * APPLICATION_EXECUTABLE     is now:   \"${APPLICATION_EXECUTABLE}\"")
+    message(NOTICE " * APPLICATION_ICON_NAME      is now:   \"${APPLICATION_ICON_NAME}\"")
+    message(NOTICE " * APPLICATION_REV_DOMAIN     is now:   \"${APPLICATION_REV_DOMAIN}\"")
+    message(NOTICE " * LINUX_PACKAGE_SHORTNAME    is now:   \"${LINUX_PACKAGE_SHORTNAME}\"")
+    message(NOTICE " * LINUX_APPLICATION_ID       is now:   \"${LINUX_APPLICATION_ID}\"")
+    message(NOTICE "")
+endif()
```

Comment:
Why not add a variable like `DEV_SUFFIX` and then just:
```cmake
if (DEV_BUILD)
set(APPLICATION_NAME "{APPLICATION_NAME}-${DEV_SUFFIX}")
....
endif()
```
This would involve a lot less changes and will be in the future easier to maintain. For daily build we then set `DEV_BUILD` to daily. We would then also use the same icon for daily and dev builds but that is fine. We don't need so much complexity:)
Also I don't would throw so much messages. Just message "Build dev version" or something similar. That should be enough.

Reply:
I could easily comment out the messages printing all of the suffixed variables. They're mostly there for debugging the changes I've made.

I had previously used a variable for the suffix, but I wanted it to be formatted differently for different strings. I could accomplish the same thing with string transformations, though.

Reply:
I don't would add so much complexity. If we really need a few different suffixes then add just a few different variables for that.